app/assets/javascripts/micropost.js

```diff
@@ -0,0 +1,8 @@
+$(document).ready(function() {
+  $('#micropost_picture').bind('change', function() {
+    var size_in_megabytes = this.files[0].size/1024/1024;
```

Comment:
thêm dấu cách trước sau toán tử đi e

Reply:
chỗ này e không hiểu ạ. a nói rõ hơn đc k ạ?

Reply:
phân tích cho a các thành phần của câu lệnh trên đi.

Reply:
phân tích cho a các thành phần của câu lệnh trên đi.

Reply:
$(document).ready(function(): sự kiện của jquery. mọi thứ trong nó sẽ tải đc ngay khi DOM đc load và trước nội dung đang đc đăng tải
DOM: document object model. mô hình các các đối tượng trong html. 
![image](https://user-images.githubusercontent.com/22794404/36773012-2ae2d456-1c8a-11e8-9a2e-b2b583319692.png) các thẻ html sẽ dc quản lí trong document (DOM)
$('#micropost_picture').bind('change', function() { 
#micropost_picture là 1 css id. theo e hiểu là bind(change) dùng để gán 1 xử lí sự kiện khi có sự thay đổi hoặc kích hoạt sự kiện đó trên 1 phần tử
    var size_in_megabytes = this.files[0].size/1024/1024;
khởi tạo biến size_in_megabytes với size của file này chia cho 1024, r chia tiếp 1024
    if (size_in_megabytes > 5) {
      alert(I18n.t('maximum_upload'));
    }
nếu size_in_megabytes >5 thì thông báo file lớn hơn 5mb chọn file nhỏ hơn ạ.

Reply:
à e hiểu r. ý là bảo chỗ size/1024/1024 cách ra thành size / 1024 / 1024 đúng k ạ :D

Reply:
mẹ ơi, tôi chỉ bảo phân tích các thành phần của câu lệnh `var size_in_megabytes = this.files[0].size/1024/1024;` thôi mẹ ạ. haiz...

Reply:
xem files này có dung lượng là bao nhiêu bytes. sau đó chia cho 1024 rồi chia tiếp cho 1024 để cho ra file đó thành megabytes
nếu nó lớn hơn 5 thì báo lỗi ạ